syntax= "proto3";

package MQTTFunction;

service MQTTManager {

rpc PublishMessages (ComingData) returns (ReturnType) {}

}

enum state {
        PublishSuccess=0;
        PublishFailed=1;
        Connexion_Lost=2;
        Client_NOT_Connected=5;
        Client_Connected=8;
        Incorrect_Proto_Version=9;
        Bad_Client_ID=10;
        Server_Unvailable=11;
        Bad_User_Pw=12;
        Not_AUTH=13;
        }

message ComingData {

    string Topic=1;
    string Payload=2;
    int32  Qos=3;
    
}

message ReturnType {
    
    
    state Acknowledgment=1;
}


